0.12 Test Runner - Plan shows Destroy/Create but not which field caused it

### Terraform Version
```
Terraform Core 247faca59d9d (master a few days ago) vendored into the Google provider
```

### Terraform Configuration Files
```hcl
resource "google_compute_network" "container_network" {
	name = "container-net"
	auto_create_subnetworks = false
}

resource "google_compute_subnetwork" "container_subnetwork" {
	name		  = "${google_compute_network.container_network.name}"
	network		  = "${google_compute_network.container_network.name}"
	ip_cidr_range = "10.128.0.0/9"
	region		  = "us-central1"
}

resource "google_container_cluster" "with_ip_allocation_policy" {
	name	   = "cluster"
	zone	   = "us-central1-a"
	network    = "${google_compute_network.container_network.name}"
	subnetwork = "${google_compute_subnetwork.container_subnetwork.name}"

	initial_node_count = 1
	ip_allocation_policy {
		use_ip_aliases           = true
		cluster_ipv4_cidr_block  = "10.0.0.0/16"
		services_ipv4_cidr_block = "10.1.0.0/16"
		node_ipv4_cidr_block     = "10.2.0.0/16"
	}
}
```

### Debug Output

https://gist.github.com/rileykarson/c673c11b3548b8f498d66eb61043c485


Plan:

```
------- Stdout: -------
=== RUN   TestAccContainerCluster_withIPAllocationPolicy_specificIPRanges
=== PAUSE TestAccContainerCluster_withIPAllocationPolicy_specificIPRanges
=== CONT  TestAccContainerCluster_withIPAllocationPolicy_specificIPRanges
--- FAIL: TestAccContainerCluster_withIPAllocationPolicy_specificIPRanges (611.66s)
    testing.go:568: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        DESTROY/CREATE: google_container_cluster.with_ip_allocation_policy
          additional_zones:                                     "" => "<computed>"
          additional_zones.#:                                   "0" => ""
          addons_config:                                        "" => "<computed>"
          addons_config.#:                                      "1" => ""
          addons_config.0.horizontal_pod_autoscaling.#:         "0" => ""
          addons_config.0.http_load_balancing.#:                "0" => ""
          addons_config.0.kubernetes_dashboard.#:               "1" => ""
          addons_config.0.kubernetes_dashboard.0.disabled:      "true" => ""
          addons_config.0.network_policy_config.#:              "1" => ""
          addons_config.0.network_policy_config.0.disabled:     "true" => ""
          cluster_autoscaling:                                  "" => "<computed>"
          cluster_autoscaling.#:                                "0" => ""
          cluster_ipv4_cidr:                                    "10.0.0.0/16" => "<computed>"
          description:                                          "" => ""
          enable_binary_authorization:                          "" => "<computed>"
          enable_kubernetes_alpha:                              "false" => "false"
          enable_legacy_abac:                                   "false" => "false"
          enable_tpu:                                           "" => "<computed>"
          endpoint:                                             "35.238.138.226" => "<computed>"
          id:                                                   "cluster-test-a273vnh67d" => "<computed>"
          initial_node_count:                                   "1" => "1"
          instance_group_urls:                                  "" => "<computed>"
          instance_group_urls.#:                                "1" => ""
          instance_group_urls.0:                                "https://www.googleapis.com/compute/v1/projects/hc-terraform-testing/zones/us-central1-a/instanceGroups/gke-cluster-test-a273vnh-default-pool-b0533daa-grp" => ""
          ip_allocation_policy.#:                               "1" => "1"
          ip_allocation_policy.0.cluster_ipv4_cidr_block:       "10.0.0.0/16" => "10.0.0.0/16"
          ip_allocation_policy.0.cluster_secondary_range_name:  "gke-cluster-test-a273vnh67d-pods-5d0da74c" => "<computed>"
          ip_allocation_policy.0.create_subnetwork:             "false" => ""
          ip_allocation_policy.0.node_ipv4_cidr_block:          "" => "10.2.0.0/16"
          ip_allocation_policy.0.services_ipv4_cidr_block:      "10.1.0.0/16" => "10.1.0.0/16"
          ip_allocation_policy.0.services_secondary_range_name: "gke-cluster-test-a273vnh67d-services-5d0da74c" => "<computed>"
          ip_allocation_policy.0.subnetwork_name:               "" => ""
          ip_allocation_policy.0.use_ip_aliases:                "true" => "true"
          location:                                             "us-central1-a" => "<computed>"
          logging_service:                                      "logging.googleapis.com" => "<computed>"
          maintenance_policy.#:                                 "0" => "0"
          master_auth:                                          "" => "<computed>"
          master_auth.#:                                        "1" => ""
          master_auth.0.client_certificate:                     "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUMyakNDQWNLZ0F3SUJBZ0lRWWV5U2UxdURHRVZDWXkvTUVoWURIVEFOQmdrcWhraUc5dzBCQVFzRkFEQXYKTVMwd0t3WURWUVFERXlRME9UbGlOMlpqTUMxbE5qRm1MVFF4WkRVdE9XTXhaQzFtTm1VeE56a3dOV1kzT1RRdwpIaGNOTVRrd016RTVNRFl3T1RRMldoY05NalF3TXpFM01EWXdPVFEyV2pBUk1ROHdEUVlEVlFRREV3WmpiR2xsCmJuUXdnZ0VpTUEwR0NTcUdTSWIzRFFFQkFRVUFBNElCRHdBd2dnRUtBb0lCQVFEQU91eU01Ti9sVEVoK2FMUkwKMDRCNEFobHdULzc5b1F3MnkyN0tsNklDZlhQN0xwQkt5NlhuZCtPRjRnVjFGZ0t6S0NJSzVyQkNCbHcwY09xRApwRno1aGZiZ0REaUxZUjYwZGdHY1NUZy9MQlMwYzBrd1B3R0JpQnViMWhLdVdRRENkWDFTUmpSVUtZa3ZjangyCkNIOHF5eHdxZlUxTlpjYmtnTE42Zk9iSS9aYnpLYWFtcm8raGZzbFFBU2E1KzNnbHNDckxwamZxTHRkZThqeC8KTE1ua04zRXhTZ29uVVg2Z1UxNWVtVFZ4NGVyenA2YXJwY3Bjd09kaSt0cWVXZjIvbFdhbk51SFB4REF4ZiswRApxVTBseDNwc1V2RTBGSmMvWGhDWTRwNVlYSzRoeUJJNGRDTDJrdVpDbEM0Vk0vRXdHM0V6a2Vnc0ExNWxjZmQzCkp2U3BBZ01CQUFHakVEQU9NQXdHQTFVZEV3RUIvd1FDTUFBd0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFCOUYKTXo4b3plS3MzTnZNVVZDNzZ4ZFFDb3Ura28yZExxekJLaHFHZVFndW9xTkVFd3drYjNhMnpzQUQyWjVhbSt5ZQprRTBkVlpmUi9qT1YwVnVBaEhydHBSWkRiYmxqdkRRR3liL21QTi91Ky9PR1BtNG4yWVZrL3U5aC9YYnp2RzZQCjNIWFZ4WjZzelkwaE9BQUJWQzZmWEc2NTRNVjlRUElEUFdlMks2aEI0b0tWQldNYVdSUmRWaDdPWEsxUXAwbjIKYXBuUjNrTysxUk94QlRyd2piNTVpYVpMMm1EeUZrOFpPY1ZqWUFCZFozcjV0STVYQ2svOTJaTnBoRWJsZkNJUQorekp1R2pQREFrcjhPbnJCSGJMUFBrZzBUKzd2OW1JaXRaOFhhZEJ5UmFtdmozVGhCZWEwUU92WGtGRnpuVjE4CkxycmdJWDNGQkcwalN4eGY5K1U9Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K" => ""
          master_auth.0.client_certificate_config.#:            "0" => ""
          master_auth.0.client_key:                             "LS0tLS1CRUdJTiBSU0EgUFJJVkFURSBLRVktLS0tLQpNSUlFcFFJQkFBS0NBUUVBd0Ryc2pPVGY1VXhJZm1pMFM5T0FlQUlaY0UvKy9hRU1Oc3R1eXBlaUFuMXoreTZRClNzdWw1M2ZqaGVJRmRSWUNzeWdpQ3Vhd1FnWmNOSERxZzZSYytZWDI0QXc0aTJFZXRIWUJuRWs0UHl3VXRITkoKTUQ4QmdZZ2JtOVlTcmxrQXduVjlVa1kwVkNtSkwzSThkZ2gvS3NzY0tuMU5UV1hHNUlDemVuem15UDJXOHltbQpwcTZQb1g3SlVBRW11ZnQ0SmJBcXk2WTM2aTdYWHZJOGZ5eko1RGR4TVVvS0oxRitvRk5lWHBrMWNlSHE4NmVtCnE2WEtYTURuWXZyYW5sbjl2NVZtcHpiaHo4UXdNWC90QTZsTkpjZDZiRkx4TkJTWFAxNFFtT0tlV0Z5dUljZ1MKT0hRaTlwTG1RcFF1RlRQeE1CdHhNNUhvTEFOZVpYSDNkeWIwcVFJREFRQUJBb0lCQUNWZDd1aXVmdlZnUUVGTwpVaW5oT2JwYVVoNmFka3RRT0lNWVJYZWRSVEZ0dUpOdHFEL1oyV3lteWZRWERRK0hFVXhKTTJEaUdnWWpuVjNXClMwZit0ZElOWUFHS2JwMXRhaDRUU0ZuY2c5SUgzQmg5eG83WjJJMlk2eUQrV3BCSW5kU3ZLUWJLdmRaVVlxWDEKODFuUWFnUGtDcHdaMG1LR3M4UnNYamNqdTJyN1JkVSt2WUliWkwzcnE1dUF5SDVQRUkyeGlGcVFaZ2FEbWhIVgo0Y1EwZ3p0czFwQlJOUmlIRFNHcmxoS2JTOFNtK3lFZ1p6YVhxWk9RK0d6VzRhblF3MnZlc0FPWW5DRGtndUlPClVJbUVHQnIyeUhQN3lMbVpvd3VQZy80S0Q4YUVLNFFVSmZtUjI2akVGSkRzbkJwRy91VDhCZEtOVWR4ajFvVE0KQ3YxNG5Ka0NnWUVBOXRqaWxpYVFpSnNIU1lGZVNpUXRSSXYwTE44dmVja0lydjNDNnpaWms2aEttTG9SVFAzNwprRTBmYWVBWEQ2VDFkSDFESmNwUG8zaDZYeHhIZ2RwZHMvMFBRYW8xd2xUMWY3K2dDR3dRRWVWVWwxNDBKeC9yCnFCUzBNblRwejFQekFSSklvOStlcWcyd2t1dy9mQXZ2emZTYnk3dnV1OUI3ODFVbllndlgzY1VDZ1lFQXgxdWEKMnYyWG9wL2FnYlBoeE9HYzFMb1lkOUZoZXZ2ZXpIeU9INk5ZNmtHWExrN3ZrSllKV0lYRWdOK3pUWEJaNVBQYQozYmdZRmM1dWE4a0tYalVVM3B4WjlUeW5HbDJRL2ZyZVBrZU1MV1VrK3ZrT1U4L2oxYkU2ZGNsREYrNG9vTDlUClYzSEQycjFYNzAzb1VwdCt6NytMWFppY1F5ay9VYnN3Rjd3ZWJaVUNnWUVBaDV0QVp5SktkbkZNUUlwTVpVaVkKZ2JhczdsV3h2Q0svblhxVXNTakxidjJkOVdjS2lCUmpOYlBZbFE1aFBOdnZhQTVFYkNkSE50UEJ6aXczOEZNbwo5TzhKaVpaYk1DWUNOQ3J1WEpJWkJldkpXOXFKcGhDNElqODFleDZWTDZ1N3JKUUJ5ZHpZU2kxNUdrMmsxcDJzCklLWEhUTll5WVlVNEFzWjkrelBzWTgwQ2dZRUFwZWl4bG1DSHFucVljV3JQa0hjbWVzOG8vOFpMOXZyWUw1dlUKOTdXTWtOL25QazNKQjY0bHR2c1VUeUJjYXl6K1V4RXlhMlVDL2xyZkFHcStheXFINUwrQUNoemhBSXN2Mkg3SQpLRktJOFZxRWpTMXFaQXZXRTRnOEc3eFVUdjBQNkw5eFQ0dkpNQ2gxbngxYUpDMXVPdHl3MGtMb2dPZzUzdjRHCmh5b3l0SVVDZ1lFQWpVdU9rTjBoOXhVZXl1N0hTTzJ3VjBLSVA2dXBocFBiWHhwQ2IrdU0wZG9oZFBucFUrSWcKQTUzOW0xcHhTWXJmczhqTGQvOHROQVYxUlhzd0dGYUVJaTBqRmIxSi9ONk5FZnVWaEw5ZVQ4MUg4YzBzOXJXUgpqQ3J0K2ZJV0ZrcSsvWE5pQWFHTnNPM3ltZ1QxQW13QkhBWGtPTXRabURMdDdONzRqTDczUVg4PQotLS0tLUVORCBSU0EgUFJJVkFURSBLRVktLS0tLQo=" => ""
          master_auth.0.cluster_ca_certificate:                 "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURDekNDQWZPZ0F3SUJBZ0lRZE1lSW5oc1M5MzZXb0FFSVhhWnpZREFOQmdrcWhraUc5dzBCQVFzRkFEQXYKTVMwd0t3WURWUVFERXlRME9UbGlOMlpqTUMxbE5qRm1MVFF4WkRVdE9XTXhaQzFtTm1VeE56a3dOV1kzT1RRdwpIaGNOTVRrd016RTVNRFV3T1RRMVdoY05NalF3TXpFM01EWXdPVFExV2pBdk1TMHdLd1lEVlFRREV5UTBPVGxpCk4yWmpNQzFsTmpGbUxUUXhaRFV0T1dNeFpDMW1ObVV4Tnprd05XWTNPVFF3Z2dFaU1BMEdDU3FHU0liM0RRRUIKQVFVQUE0SUJEd0F3Z2dFS0FvSUJBUURXNlNhSXpGY3Y4c0hjRGdSanBKN1pIMktZNFVVbngvUnZVMWRnbDNQTQpxYWdENHdMS09QNFVoN3JnWGh5OG1hUHFZK3EvL3YxOWN1eUNKemNGeGNhd242SnM4VisvWE5QMGJ6c1RWRmZaCkszY0FzMXJEZUFGdVRLbW9LenFUREU1c0RZeSsrYUt1Wk5xWjg5anlvNDhsbEdRS1FjYmxULzJHOU5BV2dxTEwKVitUdEtNQlBVa1h3ZEVPSEttYnZSQWdvV1k2OU1hNG9HZmU5UXVWRVgwcUlFdlR2R0lEaHN2S2N4eGc0ZjRGMgpZYXZmNnJvU3FoR29tSGtFQVprbzlyNFJaYlQ0Y1hRV3Zqbyt2emU5a2RGaDlTc1J5aGJUaWlnTXNLWFJTTGtECkl1QjVTeGtRL3ZDTTBRYi9pUTVabU05MUhTZW9BRVBwTXZCd1U3U0xQakpEQWdNQkFBR2pJekFoTUE0R0ExVWQKRHdFQi93UUVBd0lDQkRBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUEwR0NTcUdTSWIzRFFFQkN3VUFBNElCQVFDQgp3WVJWSlppbDFtZ1RwbnQrSHlGWEdpcWNHdVJVSXFwZzhaaVhTdHdHejd3SzI5dGRvY1BKLzhFQUZ4S0VzVE9vCklScjUrZ09hd2YrQytwMEdwTWY2aFlUUHEwR3hobzdESzB5ZTRzUzhManVBbUVGcG9Zb1R3WWxrNjNuVXBrMDQKSEJmRTRObTh3UWJGSHlscUd5SC9ZZkRXYVhuS0doSDFrRmRtcVR0M3VNeGw3SUl3SCs3c2tkRjE0RGZKREJ6NwpWWDZiYmw2RUl1UytZL2lxbUJRUjB2L0J2WmtLOTRlS1E5K2d0QTkrNnMzNE5JN0R2NjBkK2M4eGkyZVhsdTBMCmVGSW1ZUDExTmtqZXFDWmk0R24yRWc4OGhjRGFhNXBnaUFrMFJsa0Zka0VOWllZcVFsVk83M0czdXIzaHNQUFEKOG94WDBKMGVpMXJPSnJkNEFGcjQKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=" => ""
          master_auth.0.password:                               "Ck4HKjyNKw551ejp" => ""
          master_auth.0.username:                               "admin" => ""
          master_authorized_networks_config.#:                  "0" => "0"
          master_ipv4_cidr_block:                               "" => "<computed>"
          master_version:                                       "1.11.7-gke.4" => "<computed>"
          monitoring_service:                                   "monitoring.googleapis.com" => "<computed>"
          name:                                                 "cluster-test-a273vnh67d" => "cluster-test-a273vnh67d"
          network:                                              "projects/hc-terraform-testing/global/networks/container-net-cluster-test-a273vnh67d" => "container-net-cluster-test-a273vnh67d"
          network_policy:                                       "" => "<computed>"
          network_policy.#:                                     "1" => ""
          network_policy.0.enabled:                             "false" => ""
          network_policy.0.provider:                            "PROVIDER_UNSPECIFIED" => ""
          node_config:                                          "" => "<computed>"
          node_config.#:                                        "1" => ""
          node_config.0.disk_size_gb:                           "100" => ""
          node_config.0.disk_type:                              "pd-standard" => ""
          node_config.0.guest_accelerator.#:                    "0" => ""
          node_config.0.image_type:                             "COS" => ""
          node_config.0.local_ssd_count:                        "0" => ""
          node_config.0.machine_type:                           "n1-standard-1" => ""
          node_config.0.min_cpu_platform:                       "" => ""
          node_config.0.oauth_scopes.#:                         "6" => ""
          node_config.0.oauth_scopes.0:                         "https://www.googleapis.com/auth/servicecontrol" => ""
          node_config.0.oauth_scopes.1:                         "https://www.googleapis.com/auth/trace.append" => ""
          node_config.0.oauth_scopes.2:                         "https://www.googleapis.com/auth/monitoring" => ""
          node_config.0.oauth_scopes.3:                         "https://www.googleapis.com/auth/devstorage.read_only" => ""
          node_config.0.oauth_scopes.4:                         "https://www.googleapis.com/auth/logging.write" => ""
          node_config.0.oauth_scopes.5:                         "https://www.googleapis.com/auth/service.management.readonly" => ""
          node_config.0.preemptible:                            "false" => ""
          node_config.0.service_account:                        "default" => ""
          node_config.0.tags.#:                                 "0" => ""
          node_config.0.taint.#:                                "0" => ""
          node_config.0.workload_metadata_config.#:             "0" => ""
          node_locations:                                       "" => "<computed>"
          node_locations.#:                                     "0" => ""
          node_pool:                                            "" => "<computed>"
          node_pool.#:                                          "1" => ""
          node_pool.0.autoscaling.#:                            "0" => ""
          node_pool.0.initial_node_count:                       "1" => ""
          node_pool.0.instance_group_urls.#:                    "1" => ""
          node_pool.0.instance_group_urls.0:                    "https://www.googleapis.com/compute/v1/projects/hc-terraform-testing/zones/us-central1-a/instanceGroupManagers/gke-cluster-test-a273vnh-default-pool-b0533daa-grp" => ""
          node_pool.0.management.#:                             "1" => ""
          node_pool.0.management.0.auto_repair:                 "false" => ""
          node_pool.0.management.0.auto_upgrade:                "false" => ""
          node_pool.0.max_pods_per_node:                        "0" => ""
          node_pool.0.name:                                     "default-pool" => ""
          node_pool.0.name_prefix:                              "" => ""
          node_pool.0.node_config.#:                            "1" => ""
          node_pool.0.node_config.0.disk_size_gb:               "100" => ""
          node_pool.0.node_config.0.disk_type:                  "pd-standard" => ""
          node_pool.0.node_config.0.guest_accelerator.#:        "0" => ""
          node_pool.0.node_config.0.image_type:                 "COS" => ""
          node_pool.0.node_config.0.local_ssd_count:            "0" => ""
          node_pool.0.node_config.0.machine_type:               "n1-standard-1" => ""
          node_pool.0.node_config.0.min_cpu_platform:           "" => ""
          node_pool.0.node_config.0.oauth_scopes.#:             "6" => ""
          node_pool.0.node_config.0.oauth_scopes.0:             "https://www.googleapis.com/auth/servicecontrol" => ""
          node_pool.0.node_config.0.oauth_scopes.1:             "https://www.googleapis.com/auth/trace.append" => ""
          node_pool.0.node_config.0.oauth_scopes.2:             "https://www.googleapis.com/auth/monitoring" => ""
          node_pool.0.node_config.0.oauth_scopes.3:             "https://www.googleapis.com/auth/devstorage.read_only" => ""
          node_pool.0.node_config.0.oauth_scopes.4:             "https://www.googleapis.com/auth/logging.write" => ""
          node_pool.0.node_config.0.oauth_scopes.5:             "https://www.googleapis.com/auth/service.management.readonly" => ""
          node_pool.0.node_config.0.preemptible:                "false" => ""
          node_pool.0.node_config.0.service_account:            "default" => ""
          node_pool.0.node_config.0.tags.#:                     "0" => ""
          node_pool.0.node_config.0.taint.#:                    "0" => ""
          node_pool.0.node_config.0.workload_metadata_config.#: "0" => ""
          node_pool.0.node_count:                               "1" => ""
          node_pool.0.version:                                  "1.11.7-gke.4" => ""
          node_version:                                         "1.11.7-gke.4" => "<computed>"
          pod_security_policy_config.#:                         "0" => "0"
          private_cluster:                                      "" => "<computed>"
          private_cluster_config.#:                             "0" => "0"
          project:                                              "hc-terraform-testing" => "<computed>"
          region:                                               "" => "<computed>"
          subnetwork:                                           "projects/hc-terraform-testing/regions/us-central1/subnetworks/container-net-cluster-test-a273vnh67d" => "container-net-cluster-test-a273vnh67d"
          zone:                                                 "us-central1-a" => "us-central1-a"
```

### Expected Behavior

This test is failing on `0.11` HEAD due to changes in the GKE API; to be clear, the test isn't expected to succeed. The issue is that Terraform is showing a destroy/recreate but not which field is causing it.

```
         ip_allocation_policy.0.node_ipv4_cidr_block:          "" => "10.2.0.0/16" (forces new resource)
```



### Actual Behavior
```
          ip_allocation_policy.0.node_ipv4_cidr_block:          "" => "10.2.0.0/16"
```

### Steps to Reproduce
On the Google provider's `0.12` branch:

```
make testacc TEST=./google TESTARGS='-run=TestAccContainerCluster_withIPAllocationPolicy_specificIPRanges'
```

--- a/config/hcl2shim/paths.go
+++ b/config/hcl2shim/paths.go
@@ -251,3 +251,26 @@ func pathFromFlatmapKeySet(key string, ty cty.Type) (cty.Path, error) {
 	// set as a whole changed.
 	return nil, nil
 }
+
+// FlatmapKeyFromPath returns the flatmap equivalent of the given cty.Path for
+// use in generating legacy style diffs.
+func FlatmapKeyFromPath(path cty.Path) string {
+	var parts []string
+
+	for _, step := range path {
+		switch step := step.(type) {
+		case cty.GetAttrStep:
+			parts = append(parts, step.Name)
+		case cty.IndexStep:
+			switch ty := step.Key.Type(); {
+			case ty == cty.String:
+				parts = append(parts, step.Key.AsString())
+			case ty == cty.Number:
+				i, _ := step.Key.AsBigFloat().Int64()
+				parts = append(parts, strconv.Itoa(int(i)))
+			}
+		}
+	}
+
+	return strings.Join(parts, ".")
+}

--- a/config/hcl2shim/paths_test.go
+++ b/config/hcl2shim/paths_test.go
@@ -3,6 +3,7 @@ package hcl2shim
 import (
 	"fmt"
 	"reflect"
+	"strconv"
 	"strings"
 	"testing"
 
@@ -365,5 +366,43 @@ func TestRequiresReplace(t *testing.T) {
 		})
 
 	}
+}
 
+func TestFlatmapKeyFromPath(t *testing.T) {
+	for i, tc := range []struct {
+		path cty.Path
+		attr string
+	}{
+		{
+			path: cty.Path{
+				cty.GetAttrStep{Name: "force_new"},
+			},
+			attr: "force_new",
+		},
+		{
+			path: cty.Path{
+				cty.GetAttrStep{Name: "attr"},
+				cty.IndexStep{Key: cty.NumberIntVal(0)},
+				cty.GetAttrStep{Name: "force_new"},
+			},
+			attr: "attr.0.force_new",
+		},
+		{
+			path: cty.Path{
+				cty.GetAttrStep{Name: "attr"},
+				cty.IndexStep{Key: cty.StringVal("key")},
+				cty.GetAttrStep{Name: "obj_attr"},
+				cty.IndexStep{Key: cty.NumberIntVal(0)},
+				cty.GetAttrStep{Name: "force_new"},
+			},
+			attr: "attr.key.obj_attr.0.force_new",
+		},
+	} {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			attr := FlatmapKeyFromPath(tc.path)
+			if attr != tc.attr {
+				t.Fatalf("expected:%q got:%q", tc.attr, attr)
+			}
+		})
+	}
 }

--- a/helper/resource/testing_config.go
+++ b/helper/resource/testing_config.go
@@ -214,6 +214,7 @@ func legacyDiffComparisonString(changes *plans.Changes) string {
 	}
 	byModule := map[string]map[string]*ResourceChanges{}
 	resourceKeys := map[string][]string{}
+	requiresReplace := map[string][]string{}
 	var moduleKeys []string
 	for _, rc := range changes.Resources {
 		if rc.Action == plans.NoOp {
@@ -239,6 +240,12 @@ func legacyDiffComparisonString(changes *plans.Changes) string {
 		} else {
 			byModule[moduleKey][resourceKey].Deposed[rc.DeposedKey] = rc
 		}
+
+		rr := []string{}
+		for _, p := range rc.RequiredReplace.List() {
+			rr = append(rr, hcl2shim.FlatmapKeyFromPath(p))
+		}
+		requiresReplace[resourceKey] = rr
 	}
 	sort.Strings(moduleKeys)
 	for _, ks := range resourceKeys {
@@ -253,6 +260,8 @@ func legacyDiffComparisonString(changes *plans.Changes) string {
 
 		for _, resourceKey := range resourceKeys[moduleKey] {
 			rc := rcs[resourceKey]
+
+			forceNewAttrs := requiresReplace[resourceKey]
 
 			crud := "UPDATE"
 			if rc.Current != nil {
@@ -341,7 +350,17 @@ func legacyDiffComparisonString(changes *plans.Changes) string {
 				// at the core layer.
 
 				updateMsg := ""
-				// TODO: Mark " (forces new resource)" in updateMsg when appropriate.
+
+				// This may not be as precise as in the old diff, as it matches
+				// everything under the attribute that was originally marked as
+				// ForceNew, but should help make it easier to determine what
+				// caused replacement here.
+				for _, k := range forceNewAttrs {
+					if strings.HasPrefix(attrK, k) {
+						updateMsg = " (forces new resource)"
+						break
+					}
+				}
 
 				fmt.Fprintf(
 					&mBuf, "  %s:%s %#v => %#v%s\n",

